### PR TITLE
Feature/dao 170

### PR DESCRIPTION
**Pull request by HectorGerardoZV**

### CODE REVIEW

File: src/contractIntegration/interaction/web3Provider/Web3ProviderClam.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import Web3 from "web3";
+import { IdentityManager } from "../../../indentityManager";
+import IInteractionConfig from "../IInteractionConfig";
+import IWeb3Provider from "./IWeb3Provider";
+import { injectable, inject } from "tsyringe";
+
+@injectable()
+class Web3ProviderClam implements IWeb3Provider {
```
Comment (bowheadadmin) [inline src/contractIntegration/interaction/web3Provider/Web3ProviderClam.ts:8]: @HectorGerardoZV Please rename the class to Web3Provider same with the file name

File: src/contractIntegration/interaction/web3Provider/Web3ProviderClam.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        let abi: any;
+        let address: string = "";
+        if (interactionType === "consent") {
+            abi = this.interactionConfig.consent.abi;
+            address = this.interactionConfig.consent.address;
+        } else if (interactionType === "access") {
+            abi = this.interactionConfig.access.abi;
+            address = this.interactionConfig.access.address;
+        } else {
+            abi = this.interactionConfig.consent.abi;
```
Comment (bowheadadmin) [inline src/contractIntegration/interaction/web3Provider/Web3ProviderClam.ts:26]: @HectorGerardoZV we should throw an error if the method was not passed

File: src/contractIntegration/interaction/web3Provider/Web3ProviderClam.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            address = this.interactionConfig.consent.address;
+        }
+        const contract = new this.provider.eth.Contract(abi, address);
+        return contract.methods;
+    }
+
+    async callContractMethod(method: Function, identity: IdentityManager) {
+        return new Promise((resolve, reject) => {
+            method.call({ from: identity.address }, function (error: Error, result: boolean) {
```
Comment (bowheadadmin) [inline src/contractIntegration/interaction/web3Provider/Web3ProviderClam.ts:35]: @HectorGerardoZV please add support to pass more parameters

**Review by bowheadadmin — CHANGES_REQUESTED** [review]

File: src/contractIntegration/interaction/web3Provider/FactoryWeb3Interaction.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        this.config = config;
+    }
+
+    public setOptionWeb3Provider(option: IOption): void {
+        if (option.name.trim() === '' && option.name.trim().length === 0) {
+            throw new Error('The name must have at least one character');
+        }
+        const optionWeb3Provider = this.web3Providers.find(optionAux => optionAux.name === option.name);
+        if (optionWeb3Provider) {
+            throw new Error('This option already exists.');
```
Comment (bowheadadmin) [inline src/contractIntegration/interaction/web3Provider/FactoryWeb3Interaction.ts:40]: @HectorGerardoZV Allow devs to update an existing option

**Review by bowheadadmin — CHANGES_REQUESTED** [review]

File: src/contractIntegration/interaction/web3Provider/FactoryWeb3Interaction.ts (hunk 4 above)
Comment (bowheadadmin) [reply src/contractIntegration/interaction/web3Provider/FactoryWeb3Interaction.ts:40]: ....
setOptionWeb3Provider({name:"urlProvider", value:"http://localhost:8080"})

...
// later in the code
setOptionWeb3Provider({name:"urlProvider", value:"http://rinkeby.com"})

File: src/contractIntegration/interaction/web3Provider/nonceManager/NonceManager.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import 'reflect-metadata'
+import ICacheEngine from "./ICacheEngine";
+import { container } from "tsyringe";
+import MemoryCacheEngine from "./MemoryCacheEngine";
+import LocalStorageCacheEngine from "./LocalStorageCacheEngine";
+
+container.register("cache", MemoryCacheEngine);
```
Comment (bowheadadmin) [inline src/contractIntegration/interaction/web3Provider/nonceManager/NonceManager.ts:7]: @HectorGerardoZV Please rename this to memory

Comment (bowheadadmin) [inline src/contractIntegration/interaction/web3Provider/nonceManager/NonceManager.ts:7]: Hello @HectorGerardoZV please rename this to "memory"

**Review by bowheadadmin — CHANGES_REQUESTED** [review]

File: src/contractIntegration/interaction/web3Provider/Web3ProviderClam.ts (hunk 1 above)
Comment (HectorGerardoZV) [reply src/contractIntegration/interaction/web3Provider/Web3ProviderClam.ts]: Done

File: src/contractIntegration/interaction/web3Provider/Web3ProviderClam.ts (hunk 2 above)
Comment (HectorGerardoZV) [reply src/contractIntegration/interaction/web3Provider/Web3ProviderClam.ts]: Done

File: src/contractIntegration/interaction/web3Provider/Web3ProviderClam.ts (hunk 3 above)
Comment (HectorGerardoZV) [reply src/contractIntegration/interaction/web3Provider/Web3ProviderClam.ts]: Done

File: src/contractIntegration/interaction/web3Provider/nonceManager/NonceManager.ts (hunk 5 above)
Comment (HectorGerardoZV) [reply src/contractIntegration/interaction/web3Provider/nonceManager/NonceManager.ts]: Done

**Pull request merged**